pkg/pillar/cmd/zfsmanager/handlediskconfig.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zfsmanager
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	libzfs "github.com/bicomsystems/go-libzfs"
+	"github.com/lf-edge/eve/api/go/info"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/vault"
+	"github.com/lf-edge/eve/pkg/pillar/zfs"
+)
+
+func getVdevNameStatus(dev libzfs.VDevTree, name string) (string, libzfs.VDevStat) {
+	if dev.Type != libzfs.VDevTypeDisk {
+		for _, d := range dev.Devices {
+			retDev, retStat := getVdevNameStatus(d, name)
+			if retDev != "" {
+				return retDev, retStat
+			}
+		}
+	} else {
+		// we can receive partition number
+		devName, err := types.GetDiskNameByPartName(dev.Name)
+		if err != nil {
+			log.Errorf("failed to get disk name by part name")
```

Comment:
Log dev.Name also so that there is no ambiguity.

Reply:
Good point, modified